File: codeforces/problemset/520B.py
#TODO O(log(n))
n,m = map(int, input().split())
mx = 20005

q = []
visited = [0]*mx
distance = [0]*mx

q.append(n)
visited[n] = 1
distance[n] = 0
#BFS
while q:
    x = q[0]
    q.pop(0)
    if x > 0 and not visited[x-1]:
        visited[x-1] = 1
        distance[x-1] = distance[x]+1
        q.append(x-1)
    if x <= m and not visited[2*x]:
        visited[2*x] = 1
        distance[2*x] = distance[x]+1
        q.append(2*x)
print(distance[m])

File: codeforces/problemset/520B.cpp
// TODO O(log(n))
#include <iostream>
#include <queue>
using namespace std;
const int N = 2 * (10000+5);
int visit[N];
int steps[N];

int main() {
    int n, m;
    cin >> n >> m;
    queue<int> q;
    q.push(n);
    visit[n] = 1;
    steps[n] = 0;

    // BFS
    while(!q.empty()) {
        int s = q.front();
        q.pop();
        // from nose s we can go to s - 1 or s * 2
        if (s > 0 && !visit[s - 1]) {
            visit[s - 1]  = 1;
            steps[s - 1] = steps[s] + 1;
            q.push(s - 1);
        }
        if (s <= m && !visit[2 * s]) {
            visit[2 * s] = 1;
            steps[2 * s] = steps[s] + 1;
            q.push(2 * s);
        }
    }
    cout << steps[m] << endl;

    return 0;
}
